### DISCUSSION TITLE
decode  error

### DISCUSSION THREAD

**Discussion by gavenwan** [fastapi-users/fastapi-users#567] [Q&A] [answered]
in route/verify.py 
////
            data = jwt.decode(
                token,
                verification_token_secret,
                audience=VERIFY_USER_TOKEN_AUDIENCE,
                algorithms=[JWT_ALGORITHM],
            )
///
this cause error below:
   raise InvalidAlgorithmError("The specified alg value is not allowed")
jwt.exceptions.InvalidAlgorithmError: The specified alg value is not allowed

search on internet ,I changed to below ,no error 
///
jwt.decode("eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VyX2lkIjoiYjI2NDUyYWQtMDZmNi00YjIzLWJkNzctODI5OGNmMWIxZDcwIiwiYXVkIjoiZmFzdGFwaS11c2VyczphdXRoIiwiZXhwIjoxNjE2NjAwMTM1fQ.RjdENU9qVLKnbHQCFq1OMrgVibYmX6ZRvmdIh9gU_IE",options={'verify_signature':False})
{'user_id': 'b26452ad-06f6-4b23-bd77-8298cf1b1d70', 'aud': 'fastapi-users:auth', 'exp': 1616600135}
///

**Comment by frankie567** [comment, marked as answer]
Looks like you are passing an authentication token (audience `fastapi-users:auth`) in the verify router which expects a verify token (audience `fastapi-users:verify`). If you use a token given by the `/request-verify-token` route, it should work.

Not sure though why it raises `InvalidAlgorithmError`. Could you give me your installed version of `fastapi-users` and `pyjwt`?

PS: I'm transferring this to [Discussions](https://github.com/frankie567/fastapi-users/discussions). Please favor this channel next time unless you have a complete bug report to show.

**Reply by gavenwan** [reply]
fastapi-user version 5.1.2
the token is generated by:
--visit "localhost/docs"
--"login" try it out
--succed then copy from "access token"
--past to "Verify" try it out
--got "VERIFY_USER_BAD_TOKEN" response
--look into route/verify.py to check out why
--got problem above (decode error)

**Reply by frankie567** [reply]
The access token generated by `/login` is **not** for the verify endpoint. Use the `Authorize` button on the top right of the interactive documentation to authenticate.

**Reply by gavenwan** [reply]
thanks.
sorry to take up your time.
I got to learn more
